alpha/veneer/semver/semver.go

```diff
@@ -434,3 +440,32 @@ func MermaidChannelWriter(cfg declcfg.DeclarativeConfig, out io.Writer) error {
 	}
 	return nil
 }
+
+func withoutBuildMetadataConflict(versions *map[string]semver.Version) error {
```

Comment:
Might be able to simplify this function a bit by:
1. making a `seen := map[string]semver.Version{}` map
2. iterating the versions and for each of them, get the `base` version by setting `Build` to `nil`, then check if that `base` version's `String()` value already exists in the seen map. if so, fail. if not, add it.

Reply:
If you wanted to get really fancy, you could have `seen := map[string][]semver.Version{}` (note the slice of versions). With that, the keys would be the semver version sans `Build`, and the values would be all of the versions that share that base.

Then you could collect and report all of the dups (i.e. all of the slices that are length > 1).

Reply:
I did the former.  Ironically, I did the latter in May as part of the getBundleVersions implementation, but there were assertions that it was more complex than necessary and we iterated away from it.

Reply:
🤷‍♂️ 

I don't have strong opinions. From a UX perspective, if there are multiple duplicates, a user will have to run and re-run the veneer multiple times to discover all of them. But I think adding multiple versions such that multiple duplicates would be possible will be rare (except for maybe the initial setup of the veneer), so probably not a big deal.